tools.yml

```diff
@@ -442,18 +453,20 @@ tools:
   toolshed.g2.bx.psu.edu/repos/devteam/fastq_groomer/fastq_groomer/.*:
     mem: 7.6
   toolshed.g2.bx.psu.edu/repos/devteam/fastq_paired_end_deinterlacer/fastq_paired_end_deinterlacer/.*:
-    mem: 30.4
+    mem: 24
   toolshed.g2.bx.psu.edu/repos/devteam/fastq_paired_end_interlacer/fastq_paired_end_interlacer/.*:
     mem: 30.4
   toolshed.g2.bx.psu.edu/repos/devteam/fastq_paired_end_joiner/fastq_paired_end_joiner/.*:
     mem: 11.4
   toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/.*:
-    cores: 8
+    cores: min(max(int(input_size * 4), 1), 16)
+    mem: min(max(int(input_size * 7), 5.6), 64)
```

Comment:
I 4GB input file would already get 16 cores? That seems to be a lot. We use 3 cores and never had complains so far.

Reply:
Mainly I was just scaling 4GB/core but it certainly could be overkill. Of course the true question is whether cores scale in an inverse linear relationship with time, which we'd have to do some testing to know. Since I'm happy to allocate more cores if I have to allocate the memory anyway and it gets the job done faster.